<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Calculator</class>
 <widget class="QDialog" name="Calculator">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>508</width>
    <height>251</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Calculator</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_2">
   <item row="0" column="0" rowspan="2" colspan="2">
    <layout class="QGridLayout" name="gridLayout">
     <item row="4" column="6">
      <widget class="QPushButton" name="expr_sub">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>-</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">allbuttons</string>
       </attribute>
      </widget>
     </item>
     <item row="3" column="2">
      <widget class="QPushButton" name="func_sin">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>sin</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">allbuttons</string>
       </attribute>
      </widget>
     </item>
     <item row="2" column="2">
      <widget class="QPushButton" name="paren_end">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>)</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">allbuttons</string>
       </attribute>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QPushButton" name="pow_3">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>x³	</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">allbuttons</string>
       </attribute>
      </widget>
     </item>
     <item row="0" column="0" colspan="7">
      <widget class="QLineEdit" name="expression">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="2" column="7">
      <widget class="QPushButton" name="all_clear">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>AC</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">allbuttons</string>
       </attribute>
      </widget>
     </item>
     <item row="6" column="6">
      <widget class="QPushButton" name="expr_div">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>/</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">allbuttons</string>
       </attribute>
      </widget>
     </item>
     <item row="3" column="5">
      <widget class="QPushButton" name="num_9">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>9</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">allbuttons</string>
       </attribute>
      </widget>
     </item>
     <item row="3" column="6">
      <widget class="QPushButton" name="expr_add">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>+</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">allbuttons</string>
       </attribute>
      </widget>
     </item>
     <item row="6" column="3" colspan="2">
      <widget class="QPushButton" name="num_0">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>0</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">allbuttons</string>
       </attribute>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QPushButton" name="pow_2">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>x²</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">allbuttons</string>
       </attribute>
      </widget>
     </item>
     <item row="4" column="1">
      <widget class="QPushButton" name="func_cosh">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>cosh</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">allbuttons</string>
       </attribute>
      </widget>
     </item>
     <item row="3" column="4">
      <widget class="QPushButton" name="num_8">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>8</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">allbuttons</string>
       </attribute>
      </widget>
     </item>
     <item row="4" column="0">
      <widget class="QPushButton" name="pow_n">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>xⁿ</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">allbuttons</string>
       </attribute>
      </widget>
     </item>
     <item row="4" column="3">
      <widget class="QPushButton" name="num_4">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>4</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">allbuttons</string>
       </attribute>
      </widget>
     </item>
     <item row="3" column="3">
      <widget class="QPushButton" name="num_7">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>7</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">allbuttons</string>
       </attribute>
      </widget>
     </item>
     <item row="4" column="7">
      <widget class="QPushButton" name="pow_minus_one">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>1/x</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">allbuttons</string>
       </attribute>
      </widget>
     </item>
     <item row="4" column="5">
      <widget class="QPushButton" name="num_6">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>6</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">allbuttons</string>
       </attribute>
      </widget>
     </item>
     <item row="5" column="2">
      <widget class="QPushButton" name="func_tan">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>tan</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">allbuttons</string>
       </attribute>
      </widget>
     </item>
     <item row="1" column="0" colspan="8">
      <widget class="QLineEdit" name="result">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="font">
        <font>
         <pointsize>22</pointsize>
        </font>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="5" column="3">
      <widget class="QPushButton" name="num_1">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>1</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">allbuttons</string>
       </attribute>
      </widget>
     </item>
     <item row="5" column="1">
      <widget class="QPushButton" name="func_tanh">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>tanh</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">allbuttons</string>
       </attribute>
      </widget>
     </item>
     <item row="0" column="7">
      <widget class="QPushButton" name="backspace">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>←</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">allbuttons</string>
       </attribute>
      </widget>
     </item>
     <item row="5" column="4">
      <widget class="QPushButton" name="num_2">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>2</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">allbuttons</string>
       </attribute>
      </widget>
     </item>
     <item row="2" column="3">
      <widget class="QLabel" name="memory">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>M</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item row="2" column="5">
      <widget class="QPushButton" name="memory_recall">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>MR</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">allbuttons</string>
       </attribute>
      </widget>
     </item>
     <item row="6" column="5">
      <widget class="QPushButton" name="num_dot">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>.</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">allbuttons</string>
       </attribute>
      </widget>
     </item>
     <item row="2" column="6">
      <widget class="QPushButton" name="memory_clear">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>MC</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">allbuttons</string>
       </attribute>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QPushButton" name="paren_begin">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>(</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">allbuttons</string>
       </attribute>
      </widget>
     </item>
     <item row="5" column="5">
      <widget class="QPushButton" name="num_3">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>3</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">allbuttons</string>
       </attribute>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QPushButton" name="func_sinh">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>sinh</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">allbuttons</string>
       </attribute>
      </widget>
     </item>
     <item row="4" column="2">
      <widget class="QPushButton" name="func_cos">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>cos</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">allbuttons</string>
       </attribute>
      </widget>
     </item>
     <item row="3" column="7">
      <widget class="QPushButton" name="pow_half">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>√</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">allbuttons</string>
       </attribute>
      </widget>
     </item>
     <item row="5" column="6">
      <widget class="QPushButton" name="expr_mul">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>*</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">allbuttons</string>
       </attribute>
      </widget>
     </item>
     <item row="5" column="0">
      <widget class="QPushButton" name="factorial">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>x!</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">allbuttons</string>
       </attribute>
      </widget>
     </item>
     <item row="2" column="4">
      <widget class="QPushButton" name="memory_save">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>MS</string>
       </property>
      </widget>
     </item>
     <item row="5" column="7" rowspan="2">
      <widget class="QPushButton" name="eval">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>=</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">allbuttons</string>
       </attribute>
      </widget>
     </item>
     <item row="4" column="4">
      <widget class="QPushButton" name="num_5">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>5</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">allbuttons</string>
       </attribute>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <tabstops>
  <tabstop>expression</tabstop>
  <tabstop>backspace</tabstop>
  <tabstop>result</tabstop>
  <tabstop>pow_2</tabstop>
  <tabstop>paren_begin</tabstop>
  <tabstop>paren_end</tabstop>
  <tabstop>memory_save</tabstop>
  <tabstop>memory_recall</tabstop>
  <tabstop>memory_clear</tabstop>
  <tabstop>all_clear</tabstop>
  <tabstop>pow_3</tabstop>
  <tabstop>func_sinh</tabstop>
  <tabstop>func_sin</tabstop>
  <tabstop>num_7</tabstop>
  <tabstop>num_8</tabstop>
  <tabstop>num_9</tabstop>
  <tabstop>expr_add</tabstop>
  <tabstop>pow_half</tabstop>
  <tabstop>pow_n</tabstop>
  <tabstop>func_cosh</tabstop>
  <tabstop>func_cos</tabstop>
  <tabstop>num_4</tabstop>
  <tabstop>num_5</tabstop>
  <tabstop>num_6</tabstop>
  <tabstop>expr_sub</tabstop>
  <tabstop>pow_minus_one</tabstop>
  <tabstop>factorial</tabstop>
  <tabstop>func_tanh</tabstop>
  <tabstop>func_tan</tabstop>
  <tabstop>num_1</tabstop>
  <tabstop>num_2</tabstop>
  <tabstop>num_3</tabstop>
  <tabstop>expr_mul</tabstop>
  <tabstop>eval</tabstop>
  <tabstop>num_0</tabstop>
  <tabstop>num_dot</tabstop>
  <tabstop>expr_div</tabstop>
 </tabstops>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="allbuttons"/>
 </buttongroups>
</ui>
